src/analysis/retail/priest/shadow/modules/spells/Voidbolt.tsx

```diff
@@ -53,9 +77,73 @@ class Voidbolt extends ExecuteHelper {
     });
   }
 
+  onVBUpdate(event: UpdateSpellUsableEvent) {
+    //this adds timestamps of voidbolt spellusable updates
+    this.VB.push(event.timestamp);
+  }
+
+  onVBCast() {
+    this.castVB += 1;
+  }
+
+  //VB is an unusal spell. It is likely that using ExecuteHelper would be better than this for most spells.
+  getVB() {
+    //console.log("VoidformArray", this.VB)
+    //convert raw timestamps of spellusables to time between them
+    const diff = [];
+
+    // start at 2, so first calc is [2]-[1]
+    //because [0] = 0, [1] = enterVF timestamp, [2] is first VB timestamp.
+    for (let i = 2; i < this.VB.length; i += 1) {
+      diff[i - 2] = this.VB[i] - this.VB[i - 1];
+    }
+
+    let waste = 0;
+    const cd = [];
+    let total = 0;
+
+    //even diff is time between available and cast
+    //odd diff is time waiting on cooldown.
+
+    //the last timestamp occurs when VF ends, and VB updates to no longer be usable
+    //If VB is off cooldown at this time, we want to add this time to the waste time
+    //if VB is on cooldown at this time, we do not want to add this time into the average recharge time.
+    for (let i = 0; i < diff.length; i += 1) {
+      if (i % 2 === 0) {
+        //even
+        waste = waste + diff[i];
+      }
+      if (i % 2 !== 0 && i !== diff.length - 1) {
+        //odd and not include final if it is odd.
+        cd.push(diff[i]);
+      }
+    }
+
+    //find the average CD during this voidform
+    for (let i = 0; i < cd.length; i += 1) {
+      total = total + cd[i];
+    }
+    const averagecd = total / cd.length;
+
+    this.miss = this.miss + Math.floor(waste / averagecd); //Any remainder is not a possible cast, so it is floored.
```

Comment:
I think this is wrong if you happen to call it twice. It will then process `this.VB` twice and produce (up to) double the missed value.

EDIT: it isn't but it relies on `enterVoidform` to reset the state. i would probably reset the `this.VB` field to an empty list at the end of this method as well as a precaution.